client/restarts.go

```diff
@@ -127,10 +127,15 @@ func (r *RestartTracker) GetState() (string, time.Duration) {
 	// Hot path if no attempts are expected
 	if r.policy.Attempts == 0 {
 		r.reason = ReasonNoRestartsAllowed
-		if r.waitRes != nil && r.waitRes.Successful() {
+
+		// If the task does not restart on a successful exit code and
+		// the exit code was successful: terminate.
+		if !r.onSuccess && r.waitRes != nil && r.waitRes.Successful() {
```

Comment:
At this layer it doesn't seem to care about whether this is a batch or service job, is that something it should check for? e.g what about a batch job that has no restarts configured and it exited zero, in that case it should be marked complete and not failed..

Reply:
`onSuccess = !batch` which is super awkward and confuses me every time I look at this code so maybe I should just rename it to `isNotBatch`? or invert it to `isBatch`?

My default is to leave it be because I'm not sure if renaming would help the confusion. Let me know if you think a different name would be better!